### PR TITLE
check all maven repo before throwing exception

### LINKED ISSUE #20933: FeatureUtility only checks one Maven repository
**Describe the bug**  
Installing both Liberty features from Maven Central as well as user features hosted in a custom Maven repository, per this page:

https://openliberty.io/docs/22.0.0.3/reference/command/featureUtility-modifications.html#_define_custom_repositories

The featureUtility.properties has been placed in the WLP_HOME/etc directory with the following content:

mavenCentralMirror.url=http://someOtherHost:8081/repository/maven-central/ (proxy repo pointing to Maven central)
customRepo.url=http://localhost:8081/repository/maven-central/ (proxy repo pointing to custom repo)

Executing the following command:

featureUtility installFeature helloWorld1    --featuresBom=test.user.test.osgi:hello-bom:1.0 --verbose

The custom repository holds the user features along with the required BOMs.

The verbose output shows both repositories are configured, but the utility fails because it only checks Maven Central (the first repo in the properties file).

./featureUtility if helloWorld1 --verbose --noCache --featuresBom=test.user.test.osgi:hello-bom:1.0
[2022-04-26, 20:45:20:780 EDT] Check following assets whether they were installed or not: [helloWorld1]
[2022-04-26, 20:45:20:835 EDT] Using old resolve API
[2022-04-26, 20:45:20:854 EDT] checkAssetsNotInstalled() ignore exception: CWWKF1259E: Unable to obtain the following assets: helloWorld1. Ensure that the specified assets are valid. To find the IDs of applicable assets, run the installUtility find command.
[2022-04-26, 20:45:20:858 EDT] checkAssetsNotInstalled() cause of exception: Top level feature not resolved: resource=helloWorld1
MissingRequirement [requirementName=helloWorld1, owningResource=null]

[2022-04-26, 20:45:20:866 EDT] Initializing ...
[2022-04-26, 20:45:20:869 EDT] Environment variables: 
[2022-04-26, 20:45:20:870 EDT] FEATURE_REPO_URL: null
[2022-04-26, 20:45:20:872 EDT] FEATURE_REPO_USER: null
[2022-04-26, 20:45:20:873 EDT] FEATURE_UTILITY_MAVEN_REPOSITORIES: []
[2022-04-26, 20:45:20:874 EDT] FEATURE_REPO_PASSWORD: *********
[2022-04-26, 20:45:20:875 EDT] FEATURE_LOCAL_REPO: null
[2022-04-26, 20:45:20:878 EDT] Overriding the environment variables using featureUtility.properties
[2022-04-26, 20:45:20:879 EDT] envmap before:
[2022-04-26, 20:45:20:880 EDT] {FEATURE_REPO_URL=null, FEATURE_REPO_USER=null, FEATURE_UTILITY_MAVEN_REPOSITORIES=[], FEATURE_REPO_PASSWORD=null, FEATURE_LOCAL_REPO=null}
[2022-04-26, 20:45:20:881 EDT] printing envmap after
[2022-04-26, 20:45:20:882 EDT] {FEATURE_REPO_URL=null, FEATURE_REPO_USER=null, FEATURE_UTILITY_MAVEN_REPOSITORIES=[http://rhel8-install11.fyre.ibm.com:8081/repository/maven-central/, http://localhost:8081/repository/maven-central/], FEATURE_REPO_PASSWORD=null, FEATURE_LOCAL_REPO=null}
[2022-04-26, 20:45:20:884 EDT] additional jsons: [test.user.test.osgi:features:1.0]
[2022-04-26, 20:45:20:886 EDT] Features installed from the remote repository will not be cached locally
[2022-04-26, 20:45:20:887 EDT] JSONs required: [com.ibm.websphere.appserver.features:features:22.0.0.3, io.openliberty.features:features:22.0.0.3, test.user.test.osgi:features:1.0]
[2022-04-26, 20:45:20:889 EDT] Found the following jsons locally: [/Users/jiwoolim/.m2/repository/com/ibm/websphere/appserver/features/features/22.0.0.3/features-22.0.0.3.json]
[2022-04-26, 20:45:20:890 EDT] Could not find all json files from local directories, now downloading from Maven..
[2022-04-26, 20:45:20:894 EDT] Using 8 threads to download artifacts.
[2022-04-26, 20:45:20:895 EDT] Using temp location: /Users/jiwoolim/Downloads/wlp22003open/tmp/
[2022-04-26, 20:45:20:896 EDT] Testing connection for repository: http://rhel8-install11.fyre.ibm.com:8081/repository/maven-central/
[2022-04-26, 20:45:24:883 EDT] Response code: 200
[2022-04-26, 20:45:24:884 EDT] Connecting to the following repository: http://rhel8-install11.fyre.ibm.com:8081/repository/maven-central/
[2022-04-26, 20:45:36:170 EDT] Successfully validated MD5 checksum for file: features-22.0.0.3.json
[2022-04-26, 20:45:36:170 EDT] Using temp location: /Users/jiwoolim/Downloads/wlp22003open/tmp/
[2022-04-26, 20:45:36:171 EDT] Testing connection for repository: http://rhel8-install11.fyre.ibm.com:8081/repository/maven-central/
[2022-04-26, 20:45:36:308 EDT] Response code: 200
[2022-04-26, 20:45:36:308 EDT] Connecting to the following repository: http://rhel8-install11.fyre.ibm.com:8081/repository/maven-central/
[2022-04-26, 20:45:36:950 EDT] Could not download this json with maven coordinate: test.user.test.osgi:features:1.0
[2022-04-26, 20:45:36:951 EDT] Downloaded the following json files from remote: [/Users/jiwoolim/Downloads/wlp22003open/tmp/io/openliberty/features/features/22.0.0.3/features-22.0.0.3.json]
[2022-04-26, 20:45:36:952 EDT] action.exception.stacktrace: null
[2022-04-26, 20:45:36:952 EDT] CWWKF1409E: Unable to find the following feature JSON files locally or on the configured Maven repository: [test.user.test.osgi:features:1.0].
com.ibm.ws.install.InstallException: CWWKF1409E: Unable to find the following feature JSON files locally or on the configured Maven repository: [test.user.test.osgi:features:1.0].
	at com.ibm.ws.install.featureUtility.FeatureUtility.getJsonFiles(FeatureUtility.java:672)
	at com.ibm.ws.install.featureUtility.FeatureUtility.<init>(FeatureUtility.java:164)
	at com.ibm.ws.install.featureUtility.FeatureUtility.<init>(FeatureUtility.java:58)
	at com.ibm.ws.install.featureUtility.FeatureUtility$FeatureUtilityBuilder.build(FeatureUtility.java:808)
	at com.ibm.ws.install.featureUtility.cli.InstallFeatureAction.install(InstallFeatureAction.java:240)
	at com.ibm.ws.install.featureUtility.cli.InstallFeatureAction.execute(InstallFeatureAction.java:257)
	at com.ibm.ws.install.featureUtility.cli.InstallFeatureAction.handleTask(InstallFeatureAction.java:78)
	at com.ibm.ws.install.featureUtility.cli.FeatureAction.handleTask(FeatureAction.java:100)
	at com.ibm.ws.install.featureUtility.FeatureUtilityExecutor.main(FeatureUtilityExecutor.java:58)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at com.ibm.ws.kernel.boot.cmdline.UtilityMain.internal_main(UtilityMain.java:173)
	at com.ibm.ws.kernel.boot.cmdline.UtilityMain.main(UtilityMain.java:53)
	at com.ibm.ws.kernel.boot.cmdline.Main.main(Main.java:52)

The utility shows it never attempts to download the JSONs from the custom repository.   If the order is reversed in the file, then it fails not finding the JSONs for core Liberty features instead of the custom ones.   Once all the files are available in the local m2 repo, the utility succeeds.

**Expected behavior**  
If the user feature is not found on the first repo, it should continue to find it in the next repo listed on featureUtility.properties

### PR DESCRIPTION
**Pull request by jjiwooLim**
fixes #20933 

Manually tested following scenarios:
`repo1` has helloWorld feature and `repo2` has simpleActivator feature.
1. Add repo1 and repo2 to featureUtility.properties then install simpleActivator -> Checks repo1 first, then finds simpleActivator in repo2. 
2. Add repo1 and repo2 to featureUtility.properties then install helloWorld and simpleActivator feature -> finds helloWorld1 from repo1 and simpleActivator from repo2

### CODE REVIEW

**Comment by jjiwooLim** [comment]
#build

**Comment by jjiwooLim** [comment]
#build

File: dev/com.ibm.ws.install.featureUtility/src/com/ibm/ws/install/featureUtility/FeatureUtility.java [hunk 1]
```diff
@@ -112,7 +112,6 @@ private FeatureUtility(FeatureUtilityBuilder builder) throws IOException, Instal
 
         map = new InstallKernelMap();
         info(Messages.INSTALL_KERNEL_MESSAGES.getLogMessage("STATE_INITIALIZING"));
-        map.put("req.ol.json.coord", "io.openliberty.features");
```
Comment (bensonlatibm) [inline dev/com.ibm.ws.install.featureUtility/src/com/ibm/ws/install/featureUtility/FeatureUtility.java:115]: What does this do, and why was it removed?

File: dev/com.ibm.ws.install/src/com/ibm/ws/install/internal/ArtifactDownloader.java [hunk 2]
```diff
@@ -173,61 +184,68 @@ private String FormatPathSuffix(String path) {
 
     private String FormatUrlSuffix(String url) {
         String result = url;
-        if (!url.endsWith("/")) {
-            result += "/";
+        if (!url.endsWith(File.separator)) {
+            result += File.separator;
```
Comment (bensonlatibm) [inline dev/com.ibm.ws.install/src/com/ibm/ws/install/internal/ArtifactDownloader.java:174]: out of curiousity, why was this changed?

Comment (bensonlatibm) [inline dev/com.ibm.ws.install/src/com/ibm/ws/install/internal/ArtifactDownloader.java:174]: out of curiousity, why was this changed?

**Comment by jjiwooLim** [comment]
#build

**Comment by jjiwooLim** [comment]
#build

**Comment by jjiwooLim** [comment]
#build

**Review by bensonlatibm — APPROVED** [review]
lgtm!

File: dev/com.ibm.ws.install.featureUtility/src/com/ibm/ws/install/featureUtility/FeatureUtility.java (hunk 1 above)
Comment (jjiwooLim) [reply dev/com.ibm.ws.install.featureUtility/src/com/ibm/ws/install/featureUtility/FeatureUtility.java:115]: It was to retrieve OpenLiberty groupID, but we already have a constant variable for this in InstallKernelMap.java

File: dev/com.ibm.ws.install/src/com/ibm/ws/install/internal/ArtifactDownloader.java (hunk 2 above)
Comment (jjiwooLim) [reply dev/com.ibm.ws.install/src/com/ibm/ws/install/internal/ArtifactDownloader.java:174]: I thought url could be local maven repo and that Windows would have `\\`, but looking at the call hierchy it looks like this is always going to be remote repo url. I can change it back later

**Comment by jjiwooLim** [comment]
#libby-bot

**Comment by jjiwooLim** [comment]
Build failures are not related to this PR.

File: dev/com.ibm.ws.install/src/com/ibm/ws/install/internal/ArtifactDownloader.java (hunk 2 above)
Comment (barulheira) [reply dev/com.ibm.ws.install/src/com/ibm/ws/install/internal/ArtifactDownloader.java:174]: Just opened #21664
